src/name_core.jl

```diff
@@ -41,19 +41,25 @@ e.g `dim((:a, :b)) == (a=1, b=2)`.
 If the second `name` argument is given, them the dimension corresponding to that `name`,
 is returned.
 e.g. `dim((:a, :b), :b) == 2`
-If that `name` is not found then `0` is returned.
+If that `name` is not found then an error is thrown.
 """
-function dim(dimnames::Tuple)
+function dim(dimnames::Tuple)  # TODO: Remove this ?
     # 0-Allocations see: `@btime (()->dim((:a, :b)))()`
     ndims = length(dimnames)
     return NamedTuple{dimnames, NTuple{ndims, Int}}(1:ndims)
 end
 
-function dim(dimnames::Tuple, name::Symbol)
-    # 0-Allocations see: `@btime  (()->dim((:a, :b), :a))()`
-    this_namemap = NamedTuple{(name,), Tuple{Symbol}}((:notfound,))  # default we will overwrite
-    full_namemap = dim(dimnames)
-    dimnum = first(merge(this_namemap, full_namemap))
+Base.@pure function dim_noerror(dimnames::Tuple{Vararg{Symbol, N}}, name::Symbol) where N
+    # 0-Allocations see: @btime  (()->dim_noerror((:a, :b, :c), :c))()
+    for ii in 1:N
+        getfield(dimnames, ii) === name && return ii
+    end
+    return :notfound
```

Comment:
This should be `0`; we can check in the function below if `dimnum === 0` and I think that'll work.

Reply:
what is the advantage?
Type-stability for if not running at compile time?